pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -67,6 +67,13 @@ def test_get_dtypes_no_level_name():
     tm.assert_series_equal(expected, idx_multitype.dtypes)
 
 
+def test_get_dtypes_duplicate_level_names():
+    # Test MultiIndex.dtypes with non-unique level names (# GH45174 )
+    result = MultiIndex.from_arrays([[1], [2]], names=[1, 1]).dtypes
```

Comment:
Could you also add a test with different dtypes?

Reply:
https://github.com/johnzangwill/pandas/blob/7777a8dae5c3485d588e19ccb098af57e0e79031/pandas/tests/indexes/multi/test_get_set.py#L70